packages/flutter/lib/src/material/popup_menu_theme.dart

```diff
@@ -0,0 +1,164 @@
+  final TextStyle textStyle;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// new values.
+  PopupMenuThemeData copyWith({
+    Color color,
+    ShapeBorder shape,
+    double elevation,
+    TextStyle textStyle,
+  }) {
+    return PopupMenuThemeData(
+      color: color ?? this.color,
+      shape: shape ?? this.shape,
+      elevation: elevation ?? this.elevation,
+      textStyle: textStyle ?? this.textStyle,
+    );
+  }
+
+  /// Linearly interpolate between two popup menu themes.
+  ///
+  /// If both arguments are null, then null is returned.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static PopupMenuThemeData lerp(PopupMenuThemeData a, PopupMenuThemeData b, double t) {
+    assert(t != null);
+    if (a == null && b == null)
+      return null;
+    return PopupMenuThemeData(
+      color: Color.lerp(a?.color, b?.color, t),
+      shape: ShapeBorder.lerp(a?.shape, b?.shape, t),
+      elevation: lerpDouble(a?.elevation, b?.elevation, t),
+      textStyle: TextStyle.lerp(a?.textStyle, b?.textStyle, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      color,
+      shape,
+      elevation,
+      textStyle,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final PopupMenuThemeData typedOther = other;
+    return typedOther.elevation == elevation
+        && typedOther.color == color
+        && typedOther.shape == shape
+        && typedOther.textStyle == textStyle;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ColorProperty('color', color, defaultValue: null));
+    properties.add(DiagnosticsProperty<ShapeBorder>('shape', shape, defaultValue: null));
+    properties.add(DoubleProperty('elevation', elevation, defaultValue: null));
+    properties.add(DiagnosticsProperty<TextStyle>('text style', textStyle, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the configuration for
+/// popup menus in this widget's subtree.
+///
+/// Values specified here are used for popup menu properties that are not
+/// given an explicit non-null value.
+class PopupMenuTheme extends InheritedWidget {
+  /// Creates a popup menu theme that controls the configurations for
+  /// popup menus in its widget subtree.
+  PopupMenuTheme({
+    Key key,
+    Color color,
+    ShapeBorder shape,
+    double elevation,
+    TextStyle textStyle,
+    Widget child,
+  }) : data = PopupMenuThemeData(
+         color: color,
+         shape: shape,
+         elevation: elevation,
+         textStyle: textStyle,
+       ),
+       super(key: key, child: child);
+
+  /// The properties for descendant popup menu widgets.
+  final PopupMenuThemeData data;
+
+  /// The closest instance of this class' [data] value that encloses the given
```

Comment:
class' should be class's. English is strange.

Reply:
Done